### DISCUSSION TITLE
Any plans on expanding the features?

### DISCUSSION THREAD

**Discussion by Akallabeth13666** [EddieDover/fvtt-player-achievements#7] [General]
Hi,

Any plans to add more functionality like attaching item rewards to the achievements?
Perhaps even some modification to character statistics automatically changing when awarded?

Additionally, perhaps some integration with Monk automated tiles?

Thanks in advance for your answer.

**Comment by EddieDover** [comment]
I wanted the system to be game-system agnostic but, I could introduce optional features, depending on other installed plugins or maybe even game systems.  For example, my other plugin “Theater of the Mind” has optional sound effects support if you have Syrinscape and Midi-QoL installed.

I'm absolutely open to discussion about new features, and I'm willing to implement features as long as they're things that people find useful.

Going back to your examples, would you care to elaborate on how you see them play out? Keep in mind that achievements can be taken away as easily as given. So something like adding an item or a stat, would mean that in a perfect world we would be able to detect if someone had that item or stat increase already. It may not be an easy thing to implement.

Regarding Monk Automated Tiles, how exactly did you see that functioning?

**Comment by EddieDover** [comment]
Also I'm going to move this to a discussion until we can work out an actual programmable issue to tackle 🙂

**Comment by Akallabeth13666** [comment]
Hi,

Let me try to explain with an example.
The way I like to run my sessions is to have a lot of hidden content, I initially teach my players to search everywhere, but if they fail to do so they can never discover something special.
Let's say there is a secret door, and behind it, there is a glowing gem, the person who touches it gets explorer achievement which for instance permanently increases their perception. It would be easier to prepare stuff like that as I would not need to remember which secret gives which perk. So Monk active tiles could assign the achievement to the person who physically used the gem (clicked on it or something). Also Judging from how you designed the UI for it, it could be way easier to handle rewards than through Monk active tiles.
While it should be possible with active tiles, knowing all the attributes seems complicated for a layman.
Instead simple drag-and-drop for items and a drop-down list for stats in the achievement window would be way better.

For me, achievement stays until the entire campaign is finished.
So for another example. If we say there is a bandit group that is harassing the village you are resting in. The achievement could be connected with the bandit leader and whoever killed them receives something special, maybe even the title 'The Slayer of so and so' and that comes with +1 Charisma or a bonus to persuasion or intimidation.

**Reply by EddieDover** [reply]
I'm sorry that I didn't reply to this until now, I wasn't notified that you posted.

Thank you for the examples. I'm currently on vacation, but let me think over this and see what kind of implementation I can come up with.

**Comment by EddieDover** [comment]
I'm going to close this discussion, but I have added some API calls and hooks that should let you integrate this with other plugins.  If you'd like more specific work done, please open another discussion and I'll gladly take a look.
